libnetdata/socket/security.c

```diff
@@ -7,6 +7,14 @@ SSL_CTX *netdata_client_ctx=NULL;
 SSL_CTX *netdata_srv_ctx=NULL;
 const char *security_key=NULL;
 const char *security_cert=NULL;
+#if OPENSSL_VERSION_NUMBER >= 0x10100000L
+const char *tls_version=NULL;
+const char *tls_ciphers=NULL;
+const char *ciphers[] = {
+        "ECDHE-ECDSA-AES256-SHA:ECDHE-RSA-AES256-SHA:ECDHE-ECDSA-AES128-SHA:ECDHE-RSA-AES128-SHA",
+        "ECDHE-RSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-SHA:!aNULL:!eNULL:!EXPORT:!DES:!RC4:!MD5:!PSK:!aECDH:!EDH-DSS-DES-CBC3-SHA:!EDH-RSA-DES-CBC3-SHA:!KRB5-DES-CBC3-SHA",
+        "TLS_AES_256_GCM_SHA384:TLS_CHACHA20_POLY1305_SHA256:TLS_AES_128_GCM_SHA256" };
```

Comment:
Can you provide some info on how this list was decided on? I'm not saying it's a bad choice (though I have not yet audited it), but we _really_ should be transparent about why we're choosing defaults when it comes to potential security issues

Reply:
it was exactly the reason i assigned you @Ferroin for exactly this question 😄

Reply:
I think we may need to compute this dynamically actually based on whether or not the user has selected a specific protocol. If they have, we should restrict to only ciphers supported by that protocol (this is kind of important, since about half the known secure ciphers are TLS 1.3 only).

Reply:
This was decided using the following command:

```
/usr/bin/openssl ciphers -s -v
```

I got the output of the command and I separated according the TLS version.

Reply:
i feel it is not reliable way 🤔

Reply:
@ilyam8 is correct here, that output changes based on the exact OpenSSL version, and you end up with all ciphers enabled, which is bad from a security perspective. I'll look into putting together a list either today or tomorrow based on protocol to use for defaults.

Also, I didn't see anywhere in the code,b ut are we telling OpenSSL to prefer the server's cipher list here? We ideally should be, as most clients use insecure defaults for compatibility reasons.

Reply:
I will bring a new commit using https://www.openssl.org/docs/man1.1.1/man3/SSL_get_ciphers.html  in few minutes.

Reply:
The last commit removes cipher lists, now we are working with the OpenSSL default unless the user specifies ciphers with new options we are bringing.

Reply:
OK, with that I'm going to mark this resolved, as that's consistent with 'normal' behavior of most apps that use OpenSSL.